src/workerd/io/worker-entrypoint.c++

```diff
@@ -181,7 +183,14 @@ kj::Promise<void> WorkerEntrypoint::request(
       failOpenClient = context.getHttpClientNoChecks(IoContext::NEXT_CLIENT_CHANNEL, false,
                                                      kj::mv(cfBlobJson));
     }
-    waitUntilTasks.add(incomingRequest->drain().attach(kj::mv(incomingRequest)));
+    waitUntilTasks.add(incomingRequest->drain().attach(kj::mv(incomingRequest)).then(
+      [&context](){
+        if (isPredictableModeForTest()){
+          auto lock = context.getWorker().getIsolate().getApiIsolate().lock();
+          jsg::IsolateBase::from(lock->v8Isolate).requestGcForTesting();
+        }
+      })
+    );
```

Comment:
Nitpick: Please follow KJ style for indentation of lambda bodies, which should be the same as any other braced block, indented one stop more than the surrounding block, i.e.:

```suggestion
    waitUntilTasks.add(incomingRequest->drain().attach(kj::mv(incomingRequest))
        .then([&context](){
      if (isPredictableModeForTest()){
        auto lock = context.getWorker().getIsolate().getApiIsolate().lock();
        jsg::IsolateBase::from(lock->v8Isolate).requestGcForTesting();
      }
    }));
```

Reply:
Will do

Reply:
Done

Reply:
Actually there's another problem here, which is that `context` may be UAF because it's owned by `incomingReuqest`, which has already been dropped at this point. But we actually do want the context to be destroyed before we GC. So we need to capture a ref to the worker itself.

We can also optimize a bit by only adding the `.then()` in predictable mode.

```cpp
    auto promise = incomingRequest->drain().attach(kj::mv(incomingRequest));
    if (isPredictableModeForTest()) {
      promise = promise.then([worker = kj::atomicAddRef(context.getWorker())]() {
        auto lock = context.getWorker().getIsolate().getApiIsolate().lock();
        jsg::IsolateBase::from(lock->v8Isolate).requestGcForTesting();
      })
    };
    waitUntilTasks.add(kj::mv(promise));
```